src/vcpkg/base/messages.cpp

```diff
@@ -95,17 +96,160 @@ namespace vcpkg
 
     LocalizedString::LocalizedString(StringView data) : m_data(data.data(), data.size()) { }
     LocalizedString::LocalizedString(std::string&& data) noexcept : m_data(std::move(data)) { }
+
 }
 
 namespace vcpkg::msg
 {
-    REGISTER_MESSAGE(SeeURL);
-    REGISTER_MESSAGE(NoteMessage);
-    REGISTER_MESSAGE(WarningMessage);
-    REGISTER_MESSAGE(ErrorMessage);
-    REGISTER_MESSAGE(InternalErrorMessage);
-    REGISTER_MESSAGE(InternalErrorMessageContact);
+    namespace
+    {
+        template<class T>
+        struct ArgExample;
```

Comment:
I think this should be a variable template rather than a class template: https://github.com/BillyONeal/vcpkg-tool/commit/ad2bf58023750907b951434441b28ba41e7ad914